### DISCUSSION TITLE
How WalletConnect work with multi providers

### DISCUSSION THREAD

**Discussion by DanilKasianenko** [WalletConnect/walletconnect-monorepo#3990] [Developer Support] [answered]
Hello, 

I can't find code how WalletConnect work with multi providers 

So I read this [article](https://eips.ethereum.org/EIPS/eip-6963) and try to read code on WalletConnect and Wagmi on the GitHub 

I don't find code if we have for example 2 providers MetaMask and Rainbow and click to connect to certain provider how WalletConnect understand and request to connect to provider that I selected

**Comment by glitch-txs** [comment, marked as answer]
if you looking for a more low level implementation of EIP-6963 you can check this website https://github.com/WalletConnect/EIP6963
